### DISCUSSION TITLE
Consul support

### DISCUSSION THREAD

**Discussion by pvnick** [autobase-tech/autobase#171] [Q&A]
Hi Vitabaks,
With a few lines addition to patroni.yml.j2, support can be added for consul. Consul can actually take care of both the high-availability and the load balancing, and you get these cool urls master.psql-cluster.service.domain.com and replica.psql-cluster.service.domain.com automatically. Really simplifies the architecture. What do you think? I might be able to help with a merge request.

Thanks,
Paul

**Comment by vitabaks** [comment]
hi @pvnick 

Consul support needs to be added and I will be grateful to you for PR.

But I think it is not enough to add a few lines to the variables file, it is also necessary to add the functionality of installing the consul cluster, since this project is designed to install all components automatically (at least we try to provide it).

**Comment by vitabaks** [comment]
Related issue - https://github.com/vitabaks/postgresql_cluster/issues/85

**Comment by pvnick** [comment]
Aha, yes that makes sense. Unfortunately, for me that presents a large barrier to entry that I just don’t think I have the time for 😕

I have a colleague that’s doing all the consul clustering, while I just need to install the agents on the DB nodes and configure patroni.yml. Can you think of any way I can help besides creating a PR with all the cluster functionality you might want?

**Reply by vitabaks** [reply]
I also need to find more time for this. 
Perhaps, with the appearance of new [sponsors](https://www.patreon.com/vitabaks), I will increase the priority of this issue.
